tests/test_auth.js

```diff
@@ -56,6 +58,73 @@ describe('SmartFile Basic API client', () => {
     });
   });
 
+  it('can start a session and handle csrftoken', (done) => {
+    const api0 = nock(API_URL)
+      .post('/api/2/session/')
+      .basicAuth({
+        user: 'username',
+        pass: 'password',
+      })
+      .reply(200, '{}', { 'Set-Cookie': ['sessionid=bar', 'csrftoken=ABCD'] });
```

Comment:
Though it is not used, this API would return a `201` rather than a `200`. Delete would be `204` below